packages/office-ui-fabric-react/src/components/Toggle/Toggle.base.tsx

```diff
@@ -105,25 +106,38 @@ export class ToggleBase extends BaseComponent<IToggleProps, IToggleState> implem
         <div className={classNames.container}>
           <KeytipData keytipProps={keytipProps} ariaDescribedBy={(toggleNativeProps as any)['aria-describedby']} disabled={disabled}>
             {(keytipAttributes: any): JSX.Element => (
-              <button
-                {...toggleNativeProps}
-                {...keytipAttributes}
-                className={classNames.pill}
-                disabled={disabled}
-                id={this._id}
-                type="button"
-                role="switch" // ARIA 1.1 definition; "checkbox" in ARIA 1.0
-                ref={this._toggleButton}
-                aria-disabled={disabled}
-                aria-checked={checked}
-                aria-label={ariaLabel ? ariaLabel : badAriaLabel}
-                data-is-focusable={true}
-                onChange={this._noop}
-                onClick={this._onClick}
-                aria-labelledby={labelledById}
-              >
-                <div className={classNames.thumb} />
-              </button>
+              <>
+                <input
+                  className={classNames.input}
+                  type="checkbox"
+                  name={name}
+                  checked={checked}
+                  aria-disabled={disabled}
+                  aria-checked={checked}
+                  data-is-focusable={false}
+                  aria-labelledby={labelledById}
+                  onChange={this._noop}
+                />
```

Comment:
This is a significant change to Toggle and I think it'd be good to add more rationale to the PR description. I have a few questions:
* Why is `input` needed? Should it always be rendered by default?
* If input is not focusable, does it need `aria` attributes that are redundant with `button`'s? Should this be hidden from an ARIA perspective if it's redundant with `button`?

Reply:
An input is helpful for when using this control as an input within a form and either listening for onChange on the form or when submitting the form and being able to traverse the inputs.

The aria attributes I added were an approximation of what I expected you to want. It's no problem to remove or alter as needed. Please provide direction.

Reply:
I'd like to see if there are ways we can support `Toggles` in `form` scenarios without adding code to `Toggle` itself. Spreading this across all components would add a lot of DOM weight for just `form` scenarios. We do have a new version of `Toggle` in experiments that I think would be able to support this feature via slots render functions, allowing `form` users to add the `input` alongside default rendered content.

Reply:
I personally disagree that adding a single input per control is a lot of DOM weight. There's no downside to having an input that you're not using and it's an added feature to be able to use the control as an input.

Though I'm curious about the experiment that you mentioned. I'm not as sold about needing to implement an input each time I use a Toggle. That's already what I'm having to do.

As a compromise, at least for the time being, we could add a boolean prop such as "asInput" or "addInput" to conditionally render the input.

Reply:
The experimental version of `Toggle` uses slots, which we are adopting to help make component variants easier to implement. It would allow you to easily create a variant, such as `ToggleForm`, where you could add the `input` (where `pill` is the slot that handles rendering the current `button` input):
```tsx
const ToggleForm = createComponent(Toggle, {
  slots: {
    pill: (props, DefaultComponent) => {
      <input ... />
      <DefaultComponent { ...props } />
    }
  });
```

Reply:
Is this for v7?

For v6, could we conditionally render the input based on if the name attribute is a string?

Reply:
Our plan is for that to be released with v7.x. For now, we could conditionally render, but our main goal with slots is to be able to make components as lightweight as possible while giving consumers tools to easily create and manage their own component variants. With that in mind, I think we may want to take this back out with the new version of Toggle (or at some point in the future.) Does that sound reasonable?

Reply:
I've updated the build to conditionally render the input based on the name prop.

It would be my preference to maintain this functionality in v7. But that's up to your team to make that decision.

Reply:
I might not be understanding something, but why is a composite component [like this](https://codepen.io/slimemold/pen/GaOvbe) inadequate?

Reply:
If you want to create a new composite component, that's certainly an option. I've had to go that approach in my own project. But I didn't feel like it was something that each project ought to need to implement. I felt like it made sense to inherently allow for this functionality with the Toggle component which is why I created this PR. And it seems like with v7 and the slots approach, that may be your intent (to have everybody extend the Toggle). I just didn't know that was in the roadmap and it's not the approach that I would take.

I think it really just boils down to me thinking of this control as being a form element (since that's my use case) vs the MS/Office plan.

Reply:
I think that it makes sense to provide more support for forms overall in fabric, right now it's a bit disjointed. However I think that adding an input to toggle just to support forms might not be necessary since there are a lot of cases when it might not be needed there. Instead we could add another component that's the composite and is explicitly for forms.